aspnetcore/Controllers/ProductsController.cs

```diff
@@ -55,12 +55,12 @@ private async Task CallHttpApiAsync()
         try
         {
             response.EnsureSuccessStatusCode();
+            span?.Finish();
         }
         catch (Exception exception)
         {
+            span?.Finish(exception);

```

Comment:
Unrelated to the blocking feature. But we should be finishing the span in the `catch` block and there's a convenient overload that takes exceptions for this reason.

Reply:
nice! didn't know that